### ISSUE TITLE
importAttributesKeyword warning shown when linting JSON imports

### ISSUE THREAD

**Issue by TrevorBurnham** [un-ts/eslint-plugin-import-x#446]
ES2025 introduced the [import attributes syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with):

```js
import "foo.json" with { type: "json" };
```

When using this syntax in code linted with eslint-plugin-import-x rules like `import-x/order`, `import-x/extensions`, or `import-x/no-unresolved`, ESLint emits the following warning:

```
You are using import attributes, without specifying the desired output syntax.
Please specify the "importAttributesKeyword" generator option, whose value can be one of:
 - "with"        : `import { a } from "b" with { type: "json" };`
 - "assert"      : `import { a } from "b" assert { type: "json" };`
 - "with-legacy" : `import { a } from "b" with type: "json";`
```

I believe the fix is to add this to the Babel config:

```
generatorOpts: { importAttributesKeyword: "with" }
```

See https://github.com/rollup/rollup/issues/5742

**Comment by SukkaW** [comment]
We do not use Babel in any part of our codebase (aside from testing our compatibility with the Babel ESLint parser).

Most likely, you are using `@babel/eslint` and didn't update your `languageOptions` or parser options in your own ESLint config.

**Comment by TrevorBurnham** [comment]
Sorry for the misreport! It turns out the culprit was actually eslint-plugin-react-hooks: https://github.com/facebook/react/issues/31793 (That issue refers to `eslint-plugin-react-compiler`, but it's the same pattern in both packages. Still unfixed as of eslint-plugin-react-hooks@7.0.1.)
